DBSCE/Overview.md

```diff
@@ -0,0 +1,219 @@
+
+### DBSC(E) use cases
+
+This section expands on the [generic design](#high-level-design) to address different enterprise use cases:
+
+#### IDP is RP and Calls Public Local Key Helper
+
+This is the same use case elaborated in the high level design [above](#high-level-design). However, we have separated the IdP/RP components of the server for the enterprise use case in the below diagram.
+
+![IDPSameAsRP-CallsPublicLocalKeyHelper](./IDPSameAsRP-CallsPublicLocalKeyHelper.svg)
+
+Highlights:
+
+- Auth tokens are not mentioned in the DBSC(E) high level design, as they are optional for the consumer use case. However, for the enterprise use case, the IdP can generate the auth tokens and embed the thumbprint of the public key in the token.
+- The token can be delivered to the RP directly through an API instead of a header based response.
+- The tokens also contain the thumpbrint of the public key, and the RP can validate the thumbprint of the public key from the token.
+
+#### IDP Calls Public Local Key Helper
+
+In Enterprise, and some consumer use cases, it is a valid use case to have separate servers as [RP](#relying-party-rp) and [IdP](#identity-provider-idp). We address the use case where these are separate entities and probably from different vendors below.
+
+For easy mapping with the existing DBSC proposal, please note:
+
+- Steps 1-16 specify the key generation process for a public local key helper.
+- Steps 17-29 are [DBSC](https://github.com/wicg/dbsc), added for completeness.
+
+![IDPCallsPublicLocalKeyHelper](./IDPCallsPublicLocalKeyHelper.svg)
+
+#### IDP Calls Private Local Key Helper
+
+A special case is for enterprises that already have `well-known` Local Key Helpers, which are expected to be trusted and enabled by default in a browser. Here, since the browser can trust a given IDP (based on the URL and a policy mapping) and can trust the IDP to invoke the appropriate Local Key Helper (refer [Local key helper on Windows](./KeyGeneration.md#local-key-helper-on-windows)), there are a few optimizations that can be made to the protocol, in skipping the `nonce`(challenge) and reducing the number of round trips between the IDP and the Local Key Helper.
+
+![IDPCallsPrivateLocalKeyHelper](./IDPCallsPrivateLocalKeyHelper.svg)
+
+Highlights:
+
+- Since the IDP (a specific URL) is trusted in this use case, the browser can skip the `nonce`(challenge) and directly call the Local Key Helper.
+- Attestation Service is optional as it is assumed that the IDP is trusted and can validate the binding statement. In other words, Attestation Service is assumed to be part of the IDP in this case.
+- The headers are modeled after confidential client flow(OAuth2) implemented with [Microsoft IDP](https://learn.microsoft.com/en-us/entra/identity-platform/refresh-tokens) as an example. It is possible to replace `Microsoft` with `Okta` and achieve the same flow.
+
```

Comment:
Only if Okta does device registration. Not sure how refresh-token and confidential client is relevant here.

Reply:
This flow will be re-done with more generic headers. Can remove the RT references, since we decided to remove microsoft specific headers.